pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/api/types.go

```diff
@@ -106,6 +105,7 @@ type {{ .Resource.Kind }}Status struct {
 	// The status of each condition is one of True, False, or Unknown.
 	// +listType=map
 	// +listMapKey=type
+	// +optional
 	Conditions []metav1.Condition ` + "`" + `json:"conditions,omitempty"` + "`" + `
```

Comment:
@JoelSpeed without the markers we will see 

```
conditions: Conditions field is missing the following markers: patchStrategy=merge, patchMergeKey=type (kubeapilinter)
	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

It seems right, without those, I understand that we can end up replacing the whole list when just one condition is updated.